src/controller/upload.rs

```diff
@@ -47,13 +48,38 @@ pub async fn controller(request: Request) -> Result<Response, Error> {
     // Validate signature
     new_kv.validate()?;
 
+    let previous_arweave_id = new_kv.clone().find_last_chain_arweave(&mut conn)?;
+
+    // Try take the kvchain data upload to the arweave.
+    let arweave_document = KVChainArweaveDocument{
+        avatar: avatar.unwrap_or("".into()),
+        uuid,
+        persona: vec![],
+        platform: new_kv.platform.clone(),
+        identity: new_kv.identity.clone(),
+        patch: new_kv.patch.clone(),
+        signature: new_kv.signature.clone(),
+        created_at: new_kv.created_at,
+        signature_payload: new_kv.signature_payload.clone(),
+        previous_id: new_kv.previous_id.clone(),
+        previous_arweave_id: previous_arweave_id.clone(),
+    };
+    
+    let arweave_upload_future = tokio::spawn(async move {
+        arweave_document.upload_to_arweave().await.ok()
+    });
+
     // Valid. Insert it.
     let kv_link = new_kv.finalize(&mut conn)?;
+
     // Apply patch
-    kv_link.perform_patch(&mut conn)?;
+    kv_link.perform_patch(&mut conn, new_kv.arweave_id)?;
 
     // All done. Build response.
     let response = query_response(&mut conn, &persona.public_key)?;
+
+    new_kv.arweave_id = arweave_upload_future.await.unwrap();
```

Comment:
It still blocks `json_response` below.

What I mean is, spawning a tokio background job and leave it alone.